ls1intum/Artemis
# `Programming exercises`: Improve appearance of results and task bar

### Checklist
#### General


#### Client

### Motivation and Context


Make the Ui of programming exercises a bit more intuitive, so students don't mix up the different elements and there is a more clearer distinction.

### Description

The "show all"/"hide all results" button now has a triangle like other show/hide toggles in Artemis, making it more intuitive. Additionally it always stays at the same place and is not put at the very bottom when showing all results.
The task bar showing icons for all tasks now uses the normal Tooltips and the horizontal line through all tasks is replaced by a small title before all tasks to make it clearer what these icons are representing.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Go to the exercise
3. Look at the UI elements before submitting once and check for their understandability
4. Submit once
5. Look at the UI elements and check for their understandability
6. Submit another time, so there are at least two results
7. Look at the UI elements one last time and check for their understandability

### Review Progress






#### Code Review
#### Manual Tests

### Screenshots


Before:
![Bildschirm­foto 2023-01-12 um 17 08 27](https://user-images.githubusercontent.com/38322605/212138697-28bbca85-83dc-4a34-b5f0-1d6c13dd37a5.png)
Now:
![Bildschirm­foto 2023-01-12 um 17 05 28](https://user-images.githubusercontent.com/38322605/212138727-b9dfa979-2a59-419c-87f5-13289b82cce1.png)
